Update MPICH, PETSc, and libMesh

cticenhour opened the pull request:
This PR replaces #20358 and #19369 and contains the following:

- An update to MPICH 4.0.1. `moose-mpich` is now a top-level package that installs the conda-forge version of MPICH. This re-enables compatibility with macOS 12.3 and should alleviate maintenance burden in the future.
  - Using conda-forge MPICH also bumps the Linux compilers to GCC 10.3.0 
- An update to PETSc 3.16.5
- HDF5 will now be checked for by default in the PETSc script, and will either be used (if present) or downloaded (if not present). HDF5 will be not be enabled by default on macOS, see discussion in #20358. This also means libMesh will enable HDF5 by default.
- An update to current libMesh master branch. 

Closes #20579

fdkong committed:
Upgrade to PETSc-3.16.5

Refs #00000

cticenhour committed:
Add check in PETSC configure to see if HDF5 exists on system. If not download it using PETSc (except for Darwin)

Refs #20033

cticenhour committed:
Enable HDF5 in libmesh configure by default, but don't require it yet (#20033)

fdkong committed:
Let PETSc figure that out

LIBS="-lmpifort -lgfortran"

fdkong committed:
Update conda/petsc/build.sh

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

fdkong committed:
Added valgrind supp

Update python/TestHarness/suppressions/errors.supp

Co-authored-by: Alex Lindsay <alexlindsay239@gmail.com>

Commit:
Use conda-forge mpich

Use conda-forge's mpich packages for linux, mac, and arm.

Closes #20579

cticenhour committed:
Update March 2022 newsletter

moosebuild commented on petsc:
```diff
@@ -1 +1 @@
-Subproject commit 09da24df01e50defd94bc4f7396f866a808ecea5
+Subproject commit f855b95493736b087b8ccc16dc6c5b29bc4b5aa8
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/986237/) on 7395ebd wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/986237/) on 7395ebd wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/986237/) on 7395ebd wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

fdkong commented:
Haha, we are updating  the whole world  and testing the whole world

cticenhour commented:
It's scary! 👻

loganharbour commented on conda/mpich/build.sh:
```diff
@@ -1,99 +1,13 @@
-make -j $CORES
-make install
+TEMP_CXXFLAGS=${CXXFLAGS%%-fdebug-prefix-map*}-std=c++17
+# Finally, swap "-std=c++14" with "" to set our basic standard requirement.
+ACTIVATION_CXXFLAGS=${TEMP_CXXFLAGS/-std=c++14/}
```
Either line 10 is wrong or the comment is wrong

cticenhour replied:
The core of the comment is correct - this does substitute C++14 if it exists with an empty string. But the modified comment could be improved. I'll fix in the next round of commits.

loganharbour replied:
To future proof this, we should probably do

```
${TMP_CXXFLAGS//-std=c++[0-9]{2}/}
```

to replace _all_ `-std=c++xx` with nothing

cticenhour replied:
Oh I like that better. Will do.

cticenhour replied:
Adjusted a little, as the regex is slightly different than the standard bash pattern matching that this substitution expects, but fixed up in a revised [1cc5c0a](https://github.com/idaholab/moose/pull/20580/commits/1cc5c0af8a5d0385f03d696710d9aeecae130a18)

fdkong commented on python/TestHarness/suppressions/errors.supp:
```diff
@@ -189,3 +189,16 @@
+   fun:hypre_BoomerAMGSetup
+   fun:HYPRE_BoomerAMGSetup
+   fun:PCSetUp_HYPRE
+}
```
```
test:aqueous_equilibrium.2species: ==1397864==    at 0x4C38CE4: calloc (vg_replace_malloc.c:762)
test:aqueous_equilibrium.2species: ==1397864==    by 0xBFF1366: hypre_HostMalloc (memory.c:174)
test:aqueous_equilibrium.2species: ==1397864==    by 0xBFF1366: hypre_MAlloc_core (memory.c:320)
test:aqueous_equilibrium.2species: ==1397864==    by 0xBFF1366: hypre_CAlloc (memory.c:806)
test:aqueous_equilibrium.2species: ==1397864==    by 0xBFF0F91: hypre_IntArrayCreate (int_array.c:26)
test:aqueous_equilibrium.2species: ==1397864==    by 0xBD8DCB6: hypre_BoomerAMGCoarseParmsHost (par_coarse_parms.c:89)
test:aqueous_equilibrium.2species: ==1397864==    by 0xBD8DD6C: hypre_BoomerAMGCoarseParms (par_coarse_parms.c:155)
test:aqueous_equilibrium.2species: ==1397864==    by 0xBD7D45F: hypre_BoomerAMGSetup (par_amg_setup.c:1455)
test:aqueous_equilibrium.2species: ==1397864==    by 0xBD5B04B: HYPRE_BoomerAMGSetup (HYPRE_parcsr_amg.c:47)
test:aqueous_equilibrium.2species: ==1397864==    by 0xAF9D118: PCSetUp_HYPRE (hypre.c:418)
test:aqueous_equilibrium.2species: ==1397864==    by 0xAFB7EBF: PCSetUp (precon.c:1017)
test:aqueous_equilibrium.2species: ==1397864==    by 0xADD778A: KSPSetUp (itfunc.c:408)
test:aqueous_equilibrium.2species: ==1397864==    by 0xADD97F8: KSPSolve_Private (itfunc.c:852)
test:aqueous_equilibrium.2species: ==1397864==    by 0xADDCA06: KSPSolve (itfunc.c:1086)
```

fdkong replied:
@lindsayad  Do not understand why this suppress does not work for this case.

It is the same to me. My eyes do not work today?

fdkong replied:
https://civet.inl.gov/job/986298/

lindsayad replied:
I will take a look and refresh my memory more on suppressions.

lindsayad replied:
I used valgrind's `--gen-suppressions`. I just looks like a more specific version of the one you already had so I dunno, but it definitely matches and resolves the MOOSE test failures. I'll check on chemical reactions now

lindsayad reviewed (commented):
Oops I forgot there’s a small libMesh PR I need. https://github.com/libMesh/libmesh/pull/3192

lindsayad commented:
I love this branch name

moosebuild commented:
Job [Test timings](https://civet.inl.gov/job/986309/) on 7395ebd wanted to post the following:

View timings [here](https://mooseframework.inl.gov/docs/PRs/20580/timing)

This comment will be updated on new commits.

cticenhour committed:
M1 fixes related to PETSc 3.16 (#20580)

Also removes old build instructions for M1 packages since we now provide 
them

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/986597/) on 18a916a wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/986597/) on 18a916a wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/986597/) on 18a916a wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/986620/) on 18a916a wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/20580/coverage/xfem)

</details>



This comment will be updated on new commits.

lindsayad committed:
More specific suppression

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/986833/) on e0d11e9 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/986833/) on e0d11e9 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/986833/) on e0d11e9 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/986833/) on e0d11e9 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour commented:
Currently working on the new warnings we're seeing in the end user targets. Should have a potential fix submitted to libmesh today for one set of them, and a commit here to fix the issue with tinydir.

lindsayad commented:
are you hitting these? https://github.com/cxong/tinydir/issues/72

cticenhour commented:
Yes! Handing things off to conda-forge mpich bumped us to GCC10, which also experiences this on Linux.

moosebuild commented:
Job [Valgrind xfem](https://civet.inl.gov/job/986893/) on e0d11e9 : invalidated by @lindsayad

lindsayad commented:
Per that tinydir issue, I don't think there's actually anything wrong with their code. I think these are spurious warnings. I'd be inclined to wrap the tinydir includes with our ignore warnings idiom

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987083/) on bf6c6f2 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/20580/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/20580/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format d2540f66703beccbeaa8675a246ec5c3cdff22fd`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987083/) on bf6c6f2 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987083/) on bf6c6f2 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour commented:
@lindsayad I had the same feeling, so I am fine with that as well

@loganharbour I'll have to bring back some of our compiler dependency lines for linux, but it won't be hard to go back (as long as conda-forge doesn't stand in our way 😂 )

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987158/) on cfcd412 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987158/) on cfcd412 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987158/) on cfcd412 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour committed:
Fixup warnings (#20580)

 - Convert extra_ghost_elem_inserter away from std::interator
 - Convert infix_ostream_iterator away from std::iterator for clang and GCC > 9.
 - Ignore memory overlap and variable warnings from tinydir
 - Fixup compound assignment operator used in boolean context in PeridynamicsMesh
 - Fixup deprecated declarations error (C++17) from Eigen in LeastSquaresFitBase

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987559/) on b828e3a wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987559/) on b828e3a wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987559/) on b828e3a wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour committed:
Fixup XFEM parallel test (#20580)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987785/) on b3a2579 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/20580/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/20580/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 83ee3d4c9d789f2dc71f5c15a45b37362250681c`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987785/) on b3a2579 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987785/) on b3a2579 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987785/) on b3a2579 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987860/) on 6ad3caa wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987860/) on 6ad3caa wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/987860/) on 6ad3caa wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour commented on framework/include/utils/InfixIterator.h:
```diff
@@ -17,8 +17,19 @@
   : public std::iterator<std::output_iterator_tag, void, void, void, void>
+#endif
 {
+#if defined(__clang__) && (__GNUC__ > 9)
```
```suggestion
#if defined(__clang__) || (__GNUC__ > 9)
```
We want clang OR GCC > 9 here.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/987875/) on 6ad3caa wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/20580/site)

This comment will be updated on new commits.

Commit:
Use conda suggested compilers

Use Conda's suggested compilers when building moose-pyhit. This package
does not / should not have the same constraints as our main stack.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/988107/) on f4468d9 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

loganharbour reviewed (changes requested):
Reminder for myself to set `{PETSC/LIBMESH}_REV` in the Dockerfile

cticenhour committed:
More March 2022 news updates

cticenhour committed:
Update libmesh

- Update MetaPhysicL
  - `OpFunction` support for `DualNumber` and `SemiDynamicSparseNumberArray`, which
    means we can now do basic parallel `sum()` etc. on AD values.
- `get_nodeset_info()` to get nodeset info
- Fixes up deprecation warnings in custom iterators due to C++17
- `Poly2Tri` updates/bugfixes related to refinement
- Support for non-uniform refinement with `Poly2Tri`
- Added virtual methods for specifying index maps that we can use for defining
  the `RBParameterizedFunction` used in EIM approximations
- Update MetaPhysicL
  - Add ValueType for `std::array`, `std::vector`
  - Implement `ReplaceAlgebraicType`
  - Add tolerance to `DynamicSparseNumberBase::sparsity_trim`
  - Simplify end/begin/rend/rbegin
  - When we exceed our `DynamicStdArray` wrapper container size, we'll throw
- Make `DynamicStdArrayWrapper` member protected
- Factor out push_node, respect offset
- Allow proper constness for our MeshBase iterators
- `libmesh_not_implemented()` in all unimplemented `TypeNTensor` functions
- README: branching description, devel Civet badge
- memcpy warning fix in `TypeNTensor`
- `Poly2Tri` refinement fixes

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit a7279e6a5f0ea1f621d894c55a771197241f8708
+Subproject commit 7315cdb99555e4d774fc149b8c659f1e56955053
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/988558/) on e0b72ca wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/988558/) on e0b72ca wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/988558/) on e0b72ca wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Test timings](https://civet.inl.gov/job/988586/) on e0b72ca : invalidated by @milljm

update git on timing machine

moosebuild commented:
Job [End user mac](https://civet.inl.gov/job/988576/) on e0b72ca : invalidated by @cticenhour

conda JSON parse error

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/989048/) on 9550677 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/989048/) on 9550677 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/989448/) on 481eaaa wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour commented:
Since the Valgrind target had to be restarted and the Peacock target is disabled for now, can I get another review on what is currently here @loganharbour?

loganharbour commented on conda/petsc/meta.yaml:
```diff
@@ -1,7 +1,7 @@
+{% set build = 0 %}
 {% set strbuild = "build_" + build|string %}
-{% set version = "3.15.1" %}
+{% set version = "3.16.5" %}
```
Don't we want to set

```
  script_env:
    - MOOSE_JOBS
```

in this too?

milljm replied:
PETSc ./configure does not understand this variable. @fdkong as I recall, we did _something_ to make it understand how many cores we wished to use (so we were not oversubscribing on HPC head nodes).

loganharbour replied:
We need to add this to the configure: https://github.com/idaholab/moose/blob/a5afafb36f97de487f062f94d61ed4903a465b43/scripts/update_and_rebuild_petsc.sh#L103

fdkong replied:
We used to have this. I am not sure how that was removed ..

fdkong replied:
Never mind. Still there

cticenhour replied:
Moved the `MAKE_NP_STR` logic into `configure_petsc.sh` so that the conda and the update script can both use it. Conda now pulls `MOOSE_JOBS` from the build environment if it exists and uses it for both the libmesh and petsc builds.

loganharbour commented on modules/doc/content/newsletter/2022_03.md:
```diff
@@ -5,14 +5,101 @@ This MOOSE Newsletter edition is in progress. Please check back in April 2022
 for a complete description of all MOOSE changes.
 !alert-end!
 
+## MPICH Updated to 4.0.1, GCC to 10.3.0
```
```suggestion
## Conda MPICH Updated to 4.0.1, GCC to 10.3.0
```

loganharbour commented on modules/doc/content/newsletter/2022_03.md:
```diff
@@ -5,14 +5,101 @@ This MOOSE Newsletter edition is in progress. Please check back in April 2022
+  - `OpFunction` support for `DualNumber` and `SemiDynamicSparseNumberArray`, which
+    means we can now do basic parallel `sum()` etc. on AD values.
+
+- `get_nodeset_info()` to get nodeset info
```
```suggestion
- `get_nodeset_info()` bugfix in `MeshBase::get_info()`
```

loganharbour commented on modules/doc/content/newsletter/2022_03.md:
```diff
@@ -5,14 +5,101 @@ This MOOSE Newsletter edition is in progress. Please check back in April 2022
 
 ## libMesh-level Changes
 
+### `2022.03.18` Update
```
Most of this content should be ripped out as I'd prefer that the newsletter not be a changelog; ping @roystgnr and @lindsayad for anything important

cticenhour replied:
We do add this information regularly whenever libMesh is updated though.

cticenhour replied:
And I only added a portion of this myself. Vast majority of it came from @lindsayad

loganharbour replied:
Yes and things should only be in it if they're readable without context; right know the majority of it is not

cticenhour replied:
Eh that's fair. I'll revise it

loganharbour replied:
I’m all for information but people will ignore it in time if it’s not understandable 😬

loganharbour commented on modules/doc/content/newsletter/2022_03.md:
```diff
@@ -5,14 +5,101 @@ This MOOSE Newsletter edition is in progress. Please check back in April 2022
+- Support for non-uniform refinement with `Poly2Tri`
+- Added virtual methods for specifying index maps that we can use for defining
+  the `RBParameterizedFunction` used in EIM approximations
+- Update MetaPhysicL
```
Should be squashed with the other metaphysicl listing

loganharbour commented on scripts/configure_petsc.sh:
```diff
@@ -31,9 +31,63 @@ function configure_petsc()
+
+  # Prioritize user-set environment variables HDF5_DIR, HDF5DIR, and HDF5_ROOT,
+  # with the first taking the greatest priority
+  if [ ! -z "$HDF5_DIR" ]; then
+    echo "INFO: HDF5 installation location was set using HDF5_DIR=$HDF5_DIR"
+    HDF5_STR="--with-hdf5-dir=$HDF5_DIR"
+  elif [ ! -z "$HDF5DIR" ]; then
+    echo "INFO: HDF5 installation location was set using HDF5DIR=$HDF5DIR"
+    HDF5_STR="--with-hdf5-dir=$HDF5DIR"
+  elif [ ! -z "$HDF5_ROOT" ]; then
+    echo "INFO: HDF5 installation location was set using HDF5_ROOT=$HDF5_ROOT"
+    HDF5_STR="--with-hdf5-dir=$HDF5_ROOT"
+  fi
```
```suggestion
  if [ -n "$HDF5_DIR" ]; then
    echo "INFO: HDF5 installation location was set using HDF5_DIR=$HDF5_DIR"
    HDF5_STR="--with-hdf5-dir=$HDF5_DIR"
  elif [ -n "$HDF5DIR" ]; then
    echo "INFO: HDF5 installation location was set using HDF5DIR=$HDF5DIR"
    HDF5_STR="--with-hdf5-dir=$HDF5DIR"
  elif [ -n "$HDF5_ROOT" ]; then
    echo "INFO: HDF5 installation location was set using HDF5_ROOT=$HDF5_ROOT"
    HDF5_STR="--with-hdf5-dir=$HDF5_ROOT"
  fi
```

loganharbour commented:
Commit history (where [conda/mpich/conda_build_config.yaml](https://github.com/idaholab/moose/pull/20580/commits/481eaaa2f83d1c2e77a05a3f55d508353931b8cb#diff-7609acdf867b9dfc2a1678ea1b3acc2e1c54d7eae3a224e6171c7ddbbb29ed14) is edited in the last commit) will work with the petsc/libmesh spack builds on the build boxes

milljm commented on conda/mpich/meta.yaml:
```diff
@@ -20,57 +18,35 @@ build:
-  summary: 'A high performance widely portable implementation of the MPI standard.'
+  home: https://mooseframework.org/
+  license: LGPL 2.1
+  summary: 'Top Level packages designed to control environment variables for conda-forge mpich'
```
I wrote this quickly in my PR. Can we fix the grammar here?

cticenhour replied:
Feel free to add a new commit, or you can post the diff and I can integrate it into my "address review" commit later.

loganharbour reviewed (changes requested):
set these new hashes as `LIBMESH_REV` and `PETSC_REV` in `docker_ci/Dockerfile`

cticenhour committed:
Address review comments (#20580)

cticenhour committed:
Update Dockerfile to new PETSc and libMesh hashes (#20580)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/991557/) on 417fcc4 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/991557/) on 417fcc4 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/991557/) on 417fcc4 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/991633/) on b2384e1 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/991633/) on b2384e1 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/991633/) on b2384e1 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

loganharbour commented:
Will merge despite failures in https://civet.inl.gov/job/991929/ and https://civet.inl.gov/job/991928/; these are WIP recipes.
